### PR TITLE
using database transaction to implement gifting in tc2 

### PR DESCRIPTION
**Pull request by realswat1**
I used database transaction to implement the transaction function of our tc2 . Initially I wanted to use promise.all without the database transaction but realized that would not account for all the edge cases . 
This is the endpoint needed for the creation of a gift. 
In this test plan video , i am testing cases where the user has less funds then the gift and also cases where user has enough funds for transaction to go through:
https://www.loom.com/share/4350ca79e3c640ad904afe6ed847c26f?sid=e06a5c2b-edff-4b56-ae8b-75b2e9d63245

in the second test we used post man to get the instances of users wallet before and after transaction failed just to be sure rollback is working :
https://www.loom.com/share/315582a57add4ee09d594a330e027055?sid=4b98f0d3-9242-4781-af1a-e0a0786fc870

### CODE REVIEW

File: exposee-app/Routes/users.js [hunk 1]
```diff
@@ -44,46 +47,51 @@ router.get('/:user_id/wallet', validate_Token, async (req, res) => {
...
-    const senderWallet = await Wallet.findOne({ where: { user_id:sender_id } });
-    console.log('hey there',senderWallet);
-    const receiverWallet = await Wallet.findOne({ where: { user_id: receiver_id } });
-    console.log('hey there 2', receiverWallet);
+    const senderWallet = await Wallet.findOne({ where: { user_id: sender_id }, transaction: t });
+    const receiverWallet = await Wallet.findOne({ where: { user_id: receiver_id }, transaction: t });
 
     if (!senderWallet || !receiverWallet) {
-      return res.status(404).json({ error: 'Sender or receiver not found' });
+      await t.rollback();
```
Comment (alfonsocheng) [inline exposee-app/Routes/users.js]: this isn't a good use for rollback, returning 404 is sufficient here, because there's nothing to rollback at this point

File: exposee-app/Routes/users.js [hunk 2]
```diff
@@ -44,46 +47,51 @@ router.get('/:user_id/wallet', validate_Token, async (req, res) => {
...
+      return res.status(404).json({ error: "Sender or receiver not found" });
     }
 
     // Check if the sender has sufficient balance
     if (senderWallet.amount < amount) {
-      return res.status(400).json({ error: 'Insufficient balance' });
+      await t.rollback();
```
Comment (alfonsocheng) [inline exposee-app/Routes/users.js]: similarly if there's insufficient balance, nothing should happen, you don't need to rollback

File: exposee-app/Routes/users.js [hunk 3]
```diff
@@ -44,46 +47,51 @@ router.get('/:user_id/wallet', validate_Token, async (req, res) => {
...
       sender_id,
       receiver_id,
       amount,
       video_id,
-    });
+    }, { transaction: t });
 
-    return res.json({ message: 'Gift sent successfully' });
+    // If everything is successful, commit the transaction
+    await t.commit();
```
Comment (alfonsocheng) [inline exposee-app/Routes/users.js:82]: nice this is a good use of commit

**Review by alfonsocheng — APPROVED** [review]
Lgtm! Good job on testing the edge cases
